### ISSUE TITLE
JSONTool does not return sub arrays

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#18505]
It seems like the json tool is only able to return a 1 level json object.  If the returned json has a property that is an array, then that property is not returned.

Test code - this query should return a list of contentlets.

```
#set($query = '{
    "query" : { 
        "query_string" : {
            "query" : "+contentType:webPageContent +webPageContent.title:{{search}}*"
        } 
    },
    "size":3 ,
    "sort":{
        "moddate":"asc"
    }
}')
#set($query = $query.replace("{{search}}", "about") )

#set($headers = {})
$headers.put("Content-Type", "application/json")
$headers.put("Authorization", "Basic YWRtaW5AZG90Y21zLmNvbTphZG1pbg==")

#set($result = $json.post("https://demo.dotcms.com/api/es/search", 30000, $headers, $query))
$result
```
It seems we should deprecate the JSONObject/JSONArray and just return jackson driven maps/lists

Another problem - it seems that we do not allow null values in our JsonObject or at least when we are serializing it.  The below code fails:
```
#set($headers = {})
$headers.put("contentType", "application/json"))

#set($myjson = $json.fetch("https://dxp-bff.xxx.xx/rest/api/dxp/v1/catalog/products?categoryId=telefoane&language=RO&limit=10&offset=0&fieldName=specialPrice&fieldValue=true"))

$dotJSON.put("jsonResponse", $myjson)
```

```
[28/09/20 16:04:14:637 EEST] WARN mapper.JsonMappingExceptionMapper: No serializer found for class com.dotmarketing.util.json.JSONObject$Null and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.TreeMap["response"]->java.util.TreeMap["bundles"])
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class com.dotmarketing.util.json.JSONObject$Null and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.TreeMap["response"]->java.util.TreeMap["bundles"])
at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77) ~[jackson-databind-2.11.1.jar:2.11.1]
```

**Comment by jdotcms** [comment]
[PR](https://github.com/dotCMS/core/pull/19423/files)

**Comment by jdotcms** [comment]
For DOC:

A new config variable has been introduced:

```
jsontool.generate.jackson=true
```

In case you want to get back to the previous functionality for the generate(String s), that uses JSONObject/JSONArray, set this property as a false

**Comment by erickgonzalez** [comment]
Tested and working as it should.
It brings 2 contentlets the code posted above.

**Comment by bryanboza** [comment]
Fixed, tested on release-20.10.1 // Postgres // FF
